### DISCUSSION TITLE
Make date/time available in input query

### DISCUSSION THREAD

**Discussion by Waynenabors** [Shopify/function-examples#215] [Feedback: General Improvements]
It would be nice to have the current time or cart time available. 

We have a metafield that represents a date range would like to know if the function is running within in the range.

**Comment by DuncanUszkay1** [comment]
Hey Wayne 👋🏻 Thanks for the feedback. Just to clarify:

>We have a metafield that represents a date range would like to know if the function is running within in the range

I presume this date range doesn't include the time of day?

**Comment by DuncanUszkay1** [comment]
Also curious- does this use case require any sort of repetition? If it does, are you updating the metafield to the new date range each time?

**Reply by Waynenabors** [reply]
Use case is a customer membership period with two dates representing the start and end of their memberships period. These are currently just iso timestamps, but could easily be another date without time.

We don't update the range but we do have to check ever x hours to see if their membership range is still active and then we save that to a metafield that gets checked in the function. 

Basically we just want to know what time the function is running or the date on the cart so we know if it falls in the range we have set so that we don't have to have a cron running to update another metafield every x hours.

Does that makes sense?

**Comment by SaeedYasin** [comment]
One of our use case for needing to have date/time is to manage customer expectations. For example, for same day delivery, on every Friday either hide this shipping option or rename it to add a message in it which states it might be delayed because of delayed shipping on weekend.

**Reply by rbarreca** [reply]
Please upvote https://github.com/Shopify/function-examples/discussions/416 if you need dateTime.

**Comment by SaeedYasin** [comment]
@DuncanUszkay1 Hi there, is there any ETA on this feature? We have a bunch of merchants who are waiting for this one.

**Comment by nickwesselman** [comment]
This shipped in `2023-10` 🎉 

https://shopify.dev/changelog/new-fields-available-for-shopify-functions-input-queries

**Reply by bantikyan** [reply]
Only Date is available. We need time also.

**Reply by nickwesselman** [reply]
This is limited due to the caching requirements for Functions. You can use `dateTimeBefore`, `dateTimeAfter`, and `dateTimeBetween` to perform logic based on a specific time. If this does not meet your needs, I'd suggest opening a new feature request with a description of your use case. Thanks!

**Reply by rbarreca** [reply]
@bantikyan please upvote https://github.com/Shopify/function-examples/discussions/416.
